bi_view_editor/models/bve_view.py

```diff
@@ -285,11 +292,12 @@ def _create_sql_view(self):
         self.env.cr.execute('CREATE or REPLACE VIEW %s as (%s)', (
```

Comment:
What do you think about:


```python
from psycopg2.errors import Error, UndefinedObject

        # ...

        try:
            self.env.cr.execute(
                "CREATE or REPLACE VIEW %s as (%s)", (AsIs(view_name), AsIs(query),)
            )
        except UndefinedObject:
            raise ValidationError(_("The generated query is invalid. Is the OVER condition correct?"))
        except Error as e:
            raise ValidationError(_("The generated query is invalid:\n%s") % e.pgerror)
```

Reply:
Ahah still blaming my dear `over condition`? I know it is the weak spot because it is the only free text input in the query definition, but maybe it would be better to print also the original exception in both cases (maybe only the `except Error` should suffice).

Moreover, I rapidly checked your solution and `psycopg2.errors` is introduced in 2.8 (ref. https://www.psycopg.org/docs/errors.html) but minimal requirement is `2.7.3.1` (ref. https://github.com/odoo/odoo/blob/012e4a3eb11feabe5634ab806ec5dd2012ffda00/requirements.txt#L28)..

Anyway I still think these changes deserve a separate issue/PR

Reply:
```python
        try:
            self.env.cr.execute(
                "CREATE or REPLACE VIEW %s as (%s)", (AsIs(view_name), AsIs(query),)
            )
        except psycopg2.Error as e:
            raise ValidationError(_("The generated query is invalid:\n%s") % e.pgerror)

```

Reply:
Haven't tried that out, but looks good